crates/dekaf/src/read.rs

```diff
@@ -241,20 +241,22 @@ impl Read {
             // as offset for efficient record batch packing.
             let kafka_offset = next_offset - 1;
 
-            records.push(Record {
-                control: false,
-                headers: Default::default(),
-                key,
-                offset: kafka_offset,
-                partition_leader_epoch: 1,
-                producer_epoch: 1,
-                producer_id: producer.as_i64(),
-                sequence: kafka_offset as i32,
-                timestamp: unix_seconds as i64 * 1000 + unix_nanos as i64 / 1_000_000, // Map into millis.
-                timestamp_type: TimestampType::LogAppend,
-                transactional: false,
-                value,
-            });
+            if !is_control {
```

Comment:
@jgraettinger This feels like a particularly high-impact change. After our talk on sprint planning today I thought more about this, and finally after a bunch of testing it _seems_ to work just fine and solve the problem of emitting a null document any time there's an ack... Am I introducing incorrectness here that you can tell?

Reply:
FWIW, I first tried just setting `control: is_control`, since it looks like that is supposed to make consumers skip over that document as if it weren't there, but that broke everything so I went with this approach instead

Reply:
My hazy recollection is that:

* Materialize will query all partitions for their high water-mark.
* It expects to read through those water-marks before it release the first "update" within it's timely data-flow machinery.
* If a collection is idle (no ongoing writes), that means the last message is an ACK. If we suppress ACKs then Materialize would basically hang forever and not return results because it never reads through its expected initial high water-mark.

Please confirm this issue isn't re-opened with this change.

Reply:
Confirmed! See https://github.com/estuary/flow/pull/1516#discussion_r1763135832